src/cargo/core/compiler/context/compilation_files.rs

```diff
@@ -559,8 +559,8 @@ fn compute_metadata<'a, 'cfg>(
     if cx.is_primary_package(unit) {
         // This is primarily here for clippy. This ensures that the clippy
         // artifacts are separate from the `check` ones.
-        if let Some(proc) = &cx.bcx.build_config.primary_unit_rustc {
-            proc.get_program().hash(&mut hasher);
+        if let Some(path) = &cx.bcx.rustc.workspace_path {
+            path.hash(&mut hasher);
         }
     }
```

Comment:
This may need to change to use something different from `is_primary_package` if we want clippy to run for all workspace packages.